### ISSUE TITLE
swig/python detected a memory leak of type 'OGRFeatureShadow *', no destructor found.

### ISSUE THREAD

**Issue by monetschemist** [OSGeo/gdal#10999]
### What is the bug?

I use ogr from osgeo in a Python script, specifically reading attributes from an ESRI File GDB using OpenFileGDB:

`driver = ogr.GetDriverByName("OpenFileGDB")`

I notice, I believe since upgrading to Ubuntu 22.04 and gdal 3.8.4+dfsg-3ubuntu3, that I get this warning at the end of execution of my script:

`swig/python detected a memory leak of type 'OGRFeatureShadow *', no destructor found.`

### Steps to reproduce the issue

These are my imports:

```
# standard imports
from __future__ import print_function
import sys
import re

# import OGR
from osgeo import ogr
```

This is how I access the OpenFileGDB driver and the file itself:

```
# use OGR specific exceptions
ogr.UseExceptions()

# get the driver
driver = ogr.GetDriverByName("OpenFileGDB")

# opening the FileGDB
try:
    gdb = driver.Open(gdbDir, 0)
except (Exception, e):
    print(e, file=sys.stderr)
    sys.exit(2)
```

This is how I read the GDB file:

```
layer = gdb.GetLayer('avi_point')
for feature in layer:
```

Once I have read all the rows into a memory data structure, I close the GDB file as follows:

`del gdb`

The rest of the code inserts the memory data structure into a PostgreSQL database using psycopg2.  When the program exits, I see the message

`swig/python detected a memory leak of type 'OGRFeatureShadow *', no destructor found.`

on the console.

### Versions and provenance

`uname -a` gives:

`Linux montpellier 6.8.0-45-generic #45-Ubuntu SMP PREEMPT_DYNAMIC Fri Aug 30 12:02:04 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`

`apt list | grep gdal` gives:

```
gdal-bin/noble,now 3.8.4+dfsg-3ubuntu3 amd64 [installed,automatic]
gdal-data/noble,noble,now 3.8.4+dfsg-3ubuntu3 all [installed,automatic]
gdal-plugins/noble,now 3.8.4+dfsg-3ubuntu3 amd64 [installed,automatic]
libgdal-dev/noble 3.8.4+dfsg-3ubuntu3 amd64
libgdal-grass/noble 1:1.0.2-6build3 amd64
libgdal34t64/noble,now 3.8.4+dfsg-3ubuntu3 amd64 [installed,automatic]
libgeo-gdal-ffi-perl/noble 0.11-1build2 amd64
python3-gdal/noble,now 3.8.4+dfsg-3ubuntu3 amd64 [installed,automatic]
```

`python3 -version` gives:

`Python 3.12.3`

**Comment by rouault** [comment]
Will be fixed in upcoming 3.9.3 per #10831

**Comment by monetschemist** [comment]
Thanks very much @rouault !

It doesn't cause me any obvious difficulty as I process only one .GDB per script run, so I'm happy to wait.

Oddly when I searched for the error message, nothing turned up, and yet I can see your [10831](https://github.com/OSGeo/gdal/pull/10831) is exactly the error I receive.  Perhaps I included too much extraneous detail.
